EIPS/eip-devp2p-snappy.md

```diff
@@ -0,0 +1,192 @@
+## Preamble
+
+    EIP: 706
+    Title: DEVp2p snappy compression
+    Author: Péter Szilágyi <peter@ethereum.org>
+    Type: Standard Track
+    Category: Networking
+    Status: Final
+    Created: 2017-09-07
+
+## Abstract
+The base networking protocol (DEVp2p) used by Ethereum currently does not employ any form of compression. This results in a massive amount of bandwidth wasted in the entire network, making both initial sync as well as normal operation slower and laggier.
+
+This EIP proposes a tiny extension to the DEVp2p protocol to enable [Snappy compression](https://en.wikipedia.org/wiki/Snappy_(compression)) on all message payloads after the initial handshake. After extensive benchmarks, results show that data traffic is decreased by 60-80% for initial sync. You can find exact numbers below.
+
+## Motivation
+Synchronizing the Ethereum main network (block 4,248,000) in Geth using fast sync currently consumes 1.01GB upload and 33.59GB download bandwidth. On the Rinkeby test network (block 852,000) it's 55.89MB upload and 2.51GB download.
+
+However, most of this data (blocks, transactions) are heavily compressable. By enabling compression at the message payload level, we can reduce the previous numbers to 1.01GB upload / 13.46GB download on the main network, and 46.21MB upload / 463.65MB download on the test network.
+
+The motivation behind doing this at the DEVp2p level (opposed to eth for example) is that it would enable compression for all sub-protocols (eth, les, bzz) seamlessly, reducing any complexity those protocols might incur in trying to individually optimize for data traffic.
+
+## Specification
+Bump the advertised DEVp2p version number from `4` to `5`. If during handshake, the remote side advertises support only for version `4`, run the exact same protocol as until now.
+
+If the remote side advertises a DEVp2p version `>= 5`, inject a Snappy compression step right before encrypting the DEVp2p message during sending:
+
+ * A message consists of `{Code, Size, Payload}`
+  * Compress the original payload with Snappy and store it in the same field.
+  * Update the message size to the length of the compressed payload.
+  * Encrypt and send the message as before, oblivious to compression.
+
+Similarly to message sending, when receiving a DEVp2p v5 message from a remote node, insert a Snappy decompression step right after the decrypting the DEVp2p message:
+
+* A message consists of `{Code, Size, Payload}`
+ * Decrypt the message payload as before, oblivious to compression.
+ * Decompress the payload with Snappy and store it in the same field.
+ * Update the message size to the length of the decompressed payload.
+
+Important caveats:
+
+ * The handshake message is **never** compressed, since it is needed to negotiate the common version.
+ * Snappy framing is **not** used, since the DEVp2p protocol already message oriented.
+
```

Comment:
Also would mention as a feature that should be no concern about compressing already compressed data since Snappy can store uncompressed literals up to 4GB in size should the compressor choose so.

Reply:
Added a note.